src/services/pdfService.ts

```diff
@@ -0,0 +1,273 @@
+    uniqueOrigins,
+    uniqueDestinations,
+    uniqueNerves,
+    structureTraversed,
+    uniqueSpecies,
+    sexes,
+    connectionTypes,
+    endorgan,
+    entitiesJourney,
+    connectionDetails,
+  } = pdfRequirement;
+  const resultSummary: PDFMAKEContent = [
+    {
+      text: `Query result summary for: ${connectionOrigin} -> ${endorgan}`,
+      style: 'header',
+      bold: true,
+      fontSize: 18,
+    },
+    {
+      text: 'Results summary:',
+      bold: true,
+      style: 'subheader',
+      margin: [0, 20, 0, 0],
+      fontSize: 16,
+    },
+    {
+      text: `This information comes from ${numOfConnections} connections. `,
+      style: 'paragraph',
+      margin: [0, 15, 0, 0],
+    },
+  ];
+
+  const summaryContentMap = {
+    Origins: uniqueOrigins,
+    Terminations: uniqueDestinations,
+    Nerves: uniqueNerves,
+    'Structure traversed': structureTraversed,
+    'Identified Species': uniqueSpecies,
+    'Sex Specific connections': sexes,
+    'Conection types': connectionTypes,
+    'Organ systems': endorgan,
+  };
+
+  for (const [key, value] of Object.entries(summaryContentMap)) {
+    if (value) {
+      resultSummary.push({
+        text: [{ text: `${key}: `, bold: true }, { text: value }],
+        margin: [0, 10, 0, 0],
+      });
+    }
+  }
+
+  const connectionDetailsContent: PDFMAKEContent = [
+    {
+      text: 'Connection details:',
+      style: 'subheader',
+      bold: true,
+      margin: [0, 30, 0, 0],
+      fontSize: 16,
+    },
+    {
+      text: `${endorgan}`,
+      bold: true,
+      style: 'paragraph',
+      decoration: 'underline',
+      fontSize: 14,
+      margin: [0, 15, 0, 0],
+    },
+  ];
+  connectionDetails.map((detail) => {
+    for (const [key, value] of Object.entries(detail)) {
+      if (key === 'Knowledge Statement') {
+        connectionDetailsContent.push({
+          text: `${value}`,
+          style: 'paragraph',
+          margin: [0, 10, 0, 0],
+        });
+        continue;
+      }
+      connectionDetailsContent.push({
+        text: [{ text: `${key}`, bold: true }, { text: `: ${value}` }],
+        margin: [0, 10, 0, 0],
+      });
+    }
+    connectionDetailsContent.push({ text: '', margin: [0, 20, 0, 0] });
+  });
+
+  // Connectivity Matrix - tables
+  const row = connectionOrigin;
+  const columns = uniqueDestinations.split(', ');
+
+  // second row - is for the vias between the origin and the respective target organs
+  const emptyColumns = [];
+  const viasRow = columns.map((column, index) => {
+    const connectionDetail = entitiesJourney.flatMap((entity) => {
+      return entity.flatMap((ent) => {
+        if (ent.destinations.includes(column)) {
+          return ent.vias;
+        }
+        return [];
+      });
+    });
+    if (connectionDetail.length !== 0) {
+      return connectionDetail.join(', ');
+    }
+    emptyColumns.push(index);
+    return null;
+  });
+  // remove the columns with the null values - to make the pdf more readable.
```

Comment:
we remove the empty columns, so it is more readable when there are a lot of columns. @ddelpiano can you confirm if this is okay?